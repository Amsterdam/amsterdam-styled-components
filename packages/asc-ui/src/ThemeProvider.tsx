import * as React from 'react'
import { StyledComponents, Theme } from '@datapunt/asc-core'
import defaultTheme from './config/theme'

interface Props {
  theme?: Theme.ThemeInterface,
  children: any
}

const ASCDefaultThemeProvider = ({ theme, children }: Props) => {
  const namespacedTheme = {
    ascTheme: Object.assign({}, defaultTheme.ascTheme, theme),
  }
  return (
    <StyledComponents.ThemeProvider theme={namespacedTheme}>
      {children}
    </StyledComponents.ThemeProvider>
  )
}

export default ASCDefaultThemeProvider
