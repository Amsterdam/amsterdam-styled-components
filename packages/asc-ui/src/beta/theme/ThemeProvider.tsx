import deepMerge from 'deepmerge'
import type { FunctionComponent, ReactNode } from 'react'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascBetaTheme } from '.'
import type { Theme } from '../types'

export interface ThemeProviderProps {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: FunctionComponent<ThemeProviderProps> = ({
  theme: defaultTheme = ascBetaTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return <AscThemeProvider theme={theme}>{children}</AscThemeProvider>
}

export default ThemeProvider
