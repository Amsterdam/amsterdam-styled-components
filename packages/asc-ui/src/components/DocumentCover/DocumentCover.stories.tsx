import * as React from 'react'
import { action } from '@storybook/addon-actions'
import DocumentCover from './DocumentCover'

export default {
  title: 'Atoms/DocumentCover',

  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div style={{ padding: '40px 10px' }}>{storyFn()}</div>
    ),
  ],
}

export const DefaultState = () => (
  <DocumentCover
    imageSrc="http://lorempixel.com/400/800/sports/8/"
    onClick={action('click')}
    description="Download PDF (12MB)"
  />
)
