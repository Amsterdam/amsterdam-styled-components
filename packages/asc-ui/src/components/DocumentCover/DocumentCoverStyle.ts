import styled from '@datapunt/asc-core'
import { getColorFromTheme } from '../../utils'

const DocumentCoverStyle = styled.div<{}>`
  display: flex;
  justify-content: center;
  width: 100%;
  margin-bottom: 20px;
  background-color: ${({ theme }) =>
    getColorFromTheme(theme, 'tint', 'level2')};
`

export default DocumentCoverStyle
