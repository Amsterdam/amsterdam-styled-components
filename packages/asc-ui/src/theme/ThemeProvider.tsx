import React, { ReactNode } from 'react'
import deepMerge from 'deepmerge'
import { ThemeProvider as AscThemeProvider } from 'styled-components'
import { ascDefaultTheme } from '.'
import { Theme } from '../types'

interface Props {
  overrides?: Partial<Theme.ThemeInterface>
  theme?: Theme.ThemeInterface
  deep?: boolean
  children: ReactNode
}

const ThemeProvider: React.FC<Props> = ({
  theme: defaultTheme = ascDefaultTheme,
  overrides,
  deep = true,
  children,
}) => {
  const theme: Theme.ThemeInterface = deep
    ? deepMerge(defaultTheme, overrides || {})
    : { ...defaultTheme, ...overrides }

  return (
    <AscThemeProvider theme={theme}>
      <>{children}</>
    </AscThemeProvider>
  )
}

export default ThemeProvider
