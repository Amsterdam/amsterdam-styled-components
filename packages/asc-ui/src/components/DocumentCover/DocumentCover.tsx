import { Download } from '@amsterdam/asc-assets'
import type { FunctionComponent, MouseEvent } from 'react'
import styled from 'styled-components'
import Button from '../Button'
import Image from '../Image'
import DocumentCoverContent from './DocumentCoverContent'
import DocumentCoverStyle from './DocumentCoverStyle'

interface Props {
  imageSrc: string
  onClick: (event: MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => void
  description: string
  alt?: string
}

const ButtonStyled = styled(Button)`
  display: flex;
  justify-content: center;
`

const DocumentCoverContentStyle: FunctionComponent<Props> = ({
  imageSrc,
  onClick,
  description,
  alt = '',
  ...otherProps
}) => (
  <DocumentCoverStyle {...otherProps}>
    <DocumentCoverContent>
      <Image src={imageSrc} alt={alt} />
      <ButtonStyled variant="primary" onClick={onClick} iconLeft={<Download />}>
        {description}
      </ButtonStyled>
    </DocumentCoverContent>
  </DocumentCoverStyle>
)

export default DocumentCoverContentStyle
