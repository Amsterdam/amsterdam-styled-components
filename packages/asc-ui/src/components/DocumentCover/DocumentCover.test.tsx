import { render, fireEvent } from '@testing-library/react'
import DocumentCover from './DocumentCover'
import ThemeProvider from '../../theme/ThemeProvider'

jest.useFakeTimers()

describe('DocumentCover', () => {
  it('should trigger the download action when the button is clicked', () => {
    const mockFn = jest.fn()
    const { getByText } = render(
      <ThemeProvider>
        <DocumentCover
          imageSrc="https://data.amsterdam.nl/assets/images/amsterdam-maps.png"
          onClick={mockFn}
          description="Download PDF (12MB)"
        />
      </ThemeProvider>,
    )
    fireEvent.click(getByText('Download PDF (12MB)'))
    expect(mockFn).toHaveBeenCalled()
  })
})
