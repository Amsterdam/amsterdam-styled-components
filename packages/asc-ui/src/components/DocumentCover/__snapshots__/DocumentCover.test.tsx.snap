// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentCover should render 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  margin-bottom: 20px;
  background-color: #F5F5F5;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 20px;
  height: 20px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c5 > svg {
  width: inherit;
  height: inherit;
}

.c5 > svg rect,
.c5 > svg polygon,
.c5 > svg path {
  fill: #ffffff;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: none;
  cursor: pointer;
  font-size: 16px;
  font-weight: normal;
  padding: 0 10px 0 10px;
  -webkit-transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,background-color 0.1s ease-in-out;
  background: #004699;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 4px 10px 4px 10px;
  min-height: 36px;
  line-height: 1em;
  color: #fff;
}

.c3:focus {
  z-index: 10;
  outline-color: #FEC813;
  outline-style: solid;
  outline-offset: 0px;
  outline-width: 3px;
}

.c3:hover {
  background: #00387a;
}

.c3:after {
  content: '';
  display: block;
  min-height: inherit;
  font-size: 0;
}

.c3 > .c4:first-child {
  margin-right: 5px;
}

.c3 > .c4:last-child {
  margin-left: 5px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 32px auto;
}

.c1 > .c2 {
  max-width: 300px;
  margin-bottom: 32px;
  width: 100%;
}

.c6 {
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0;
  margin-left: 0;
  color: #000000;
  font-weight: 400;
  font-size: 14px;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 18px;
  margin-bottom: 0;
  font-stretch: normal;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  color: #ffffff;
}

@media screen and (max-width:540px) {
  .c1 {
    margin: 20px;
  }
}

@media screen and (max-width:540px) {
  .c1 > .c2 {
    max-width: 200px;
    margin-bottom: 20px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt=""
      class="c2 "
      src="https://data.amsterdam.nl/assets/images/amsterdam-maps.png"
    />
    <button
      class="c3"
      color="primary"
    >
      <span
        class="c4 c5"
        color="bright"
        size="20"
      >
        <svg
          focusable="false"
          viewBox="0 0 32 32"
        >
          <path
            d="M0 27.996h32V32H0v-4.004zm22.36-13.983l2.83 2.823L16 26.044l-9.19-9.208 2.83-2.823L14 18.377V0h4v18.377l4.36-4.364z"
          />
        </svg>
      </span>
      <span
        class="c6"
        color="bright"
      >
        Download PDF (12MB)
      </span>
    </button>
  </div>
</div>
`;
